### PR TITLE
[FIX-#4172][server-worker] kill task NPE

### LINKED ISSUE #4172: TaskKillProcessor throw NullPointException
1.when worker runing more than 100 tasks , there are some tasks in queue.
with that ,master send a kill commad to this worker,TaskKillProcessor receive ,and get processId in cache(TaskExecutionContext).
But the taskInstance isn't running in this worker ,it is in queue.so the TaskKillProcessor  throw NullPointException.
2.This exception will be catched in this method.The TaskKillProcessor will continue run to build command .In method "buildKillTaskResponseCommand" , get TaskInstanceId in cache(TaskExecutionContext),but this cache is null,so the command send to master with a null taskInstanceId.

### PR DESCRIPTION
**Pull request by CalvinKirs**
[FIX-#4172][server-worker] kill task NPE

The cache task will be sent when the Process is generated. Before that, if a kill task appears, then NPE will appear
Modification method: write into the cache when the task is received, and mark it as preData
If the task is killed before the Process is generated, delete the cache directly at this time
It will be judged before the process is generated. If the task has been killed, it will not be executed.
After the new process is created, write it into the cache, and judge again, if kill, then kill the process.

this closes #4172

### CODE REVIEW

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/AbstractCommandExecutor.java [hunk 1]
```diff
@@ -135,12 +143,17 @@ private void buildProcess(String commandFile) throws IOException {
...
-    public CommandExecuteResult run(String execCommand) throws Exception{
+    public CommandExecuteResult run(String execCommand) throws Exception {
 
         CommandExecuteResult result = new CommandExecuteResult();
 
-
+        int taskInstanceId = taskExecutionContext.getTaskInstanceId();
+        // If the task has been killed, then the task in the cache is null
+        if (null == taskExecutionContextCacheManager.getByTaskInstanceId(taskInstanceId)) {
+            result.setExitStatusCode(EXIT_CODE_KILL);
```
Comment (lenboo) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/AbstractCommandExecutor.java:153]: i think here you can return result and ignore the next process

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/AbstractCommandExecutor.java [hunk 2]
```diff
@@ -155,14 +169,17 @@ public CommandExecuteResult run(String execCommand) throws Exception{
...
 
         result.setProcessId(processId);
 
         // cache processId
         taskExecutionContext.setProcessId(processId);
-        taskExecutionContextCacheManager.cacheTaskExecutionContext(taskExecutionContext);
+        boolean updateTaskExecutionContextStatus = taskExecutionContextCacheManager.updateTaskExecutionContext(taskExecutionContext);
+        if (Boolean.FALSE.equals(updateTaskExecutionContextStatus)) {
+            ProcessUtils.kill(taskExecutionContext);
+            result.setExitStatusCode(EXIT_CODE_KILL);
```
Comment (lenboo) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/AbstractCommandExecutor.java:181]: whether here need return the result too?

Comment (CalvinKirs) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/AbstractCommandExecutor.java:181]: Thx.done

**Review by lenboo — APPROVED** [review]
+1

**Comment by zhuangchong** [comment]
LGTM.

**Review by lgcareer — APPROVED** [review]
+1

**Review by lgcareer — APPROVED** [review]
+1

**Pull request merged**